source/slang/slang-ir-legalize-types.cpp

```diff
@@ -2046,6 +2051,25 @@ static LegalVal coerceToLegalType(IRTypeLegalizationContext* context, LegalType
     }
 }
 
+static LegalVal legalizeUndefined(IRTypeLegalizationContext* context, IRInst* inst)
+{
+    if (auto structType = as<IRStructType>(inst->getFullType()))
+    {
+        for (auto field : structType->getFields())
+        {
+            if (isResourceType(field->getFieldType()))
```

Comment:
I think you want to define a isOpaqueType() functions that returns true for handles and structs that contains opaque type fields. This logic won’t work if there are two levels of nesting.